docs/ci-cd-environment/docker-authentication.md

```diff
@@ -0,0 +1,157 @@
+Please check instructions on [how to authenticate Docker pulls](#how-to-authenticate-Docker-pulls) in the section below.  
+
+## How can you know if you are hitting the limit
+If you have exceeded the rate limit Docker will throw the `Too Many Requests` error.  
+In the job log check the output of your `docker pull` command, if you have exceeded the rate limit output will be the following:  
+```bash
+Error pulling image <docker-image-name>: Error response from daemon: toomanyrequests: Too Many Requests.`
+```
+
+## How to authenticate Docker pulls
+### Create the Semaphore secret  
+The first step is to store your Docker Hub credentials. You can use [Semaphore secret](https://docs.semaphoreci.com/essentials/using-secrets/) to safely store any credentials and make them available in your projects.  
+
+**Creating a secret from the UI**
+- Click on the organization icon in the top right corner  
+- From the menu select **Settings**  
+- On the left side pick **Secrets**  
+- Click on **New Secret**  
+- Fill in a unique name for your secret  
+- Add the first environment variable: `Variable name: "DOCKER_CREDENTIAL_TYPE", Value: "DockerHub"`  
+- Click on **+ Add another** and add new variable: `Variable name: DOCKER_USERNAME, Value:<your-dockerhub-username>`  
+- Add the third environment variable: `Variable name: DOCKER_PASSWORD, Value:<your-dockerhub-password>`  
+- Click on **Save Secret**  
+
+**Creating a secret through CLI**  
+Before you begin, you'll need to [install the Semaphore CLI][install-cli].  
+
+After connecting to your Semaphore organization update the details in the example command below and run it:  
+```bash
+sem create secret <name-of-your-secret> \
+  -e DOCKER_CREDENTIAL_TYPE=DockerHub \
+  -e DOCKER_USERNAME=<your-dockerhub-username> \
+  -e DOCKER_PASSWORD=<your-dockerhub-password> \
+```
+**Adding a secret to your pipeline YAML**  
+To use your newly created secret in your jobs you have to first attach it.  
+You can attach a secret to individual blocks in your workflow or the whole pipeline.  
+We suggest doing the latter so that it's available to **all jobs** in the workflow.  
+You can achieve this by using `global_job_config` like this:  
+```yaml
+version: v1.0
+name: My project
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+global_job_config:
+  # Connect secret to all jobs in the pipeline
+  secrets:
+    - name: <your-docker-hub-secret>
+
+blocks:
+  ...
+```
+
+### Use secret to authenticate Docker images pulls  
+For your docker pulls to be authenticated you have to log into Docker Hub:  
+```bash
+echo $DOCKER_PASSWORD | docker login --username "$DOCKER_USERNAME" --password-stdin
+```
+You should run this command before pulling any Docker images.  
+
+Same as with secrets, to avoid having to add this to every job or block we suggest that you include it in the `global_job_config` prologue of your pipeline:
```

Comment:
Same as with secrets, to avoid having to add this to every job or block we suggest that => Same as with secrets, to avoid having to add this to every job or block, we suggest that